### DISCUSSION TITLE
crash while playing on fabric

### DISCUSSION THREAD

**Discussion by rdyEmber** [FabricMC/fabric-api#2744] [Player support]
when i was playing with a lot of mods (had 30 FPS) - suddenly my MC crashed and idk why it crashed
The game crashed whilst unexpected error
Error: java.lang.ClassCastException: class net.minecraft.class_746 cannot be cast to class net.minecraft.class_3222 (net.minecraft.class_746 and net.minecraft.class_3222 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @314c508a)
and heres pastebin: https://pastebin.com/JzgF98zt

**Comment by Juuxel** [comment]
Appears to be a Spectrum issue (or a Spectrum/Croptopia compat issue?).

(The unobfuscated error message is "class ClientPlayerEntity cannot be cast to class ServerPlayerEntity")

**Comment by rdyEmber** [comment]
oh
let me guess, i have to delete both mods or one of those?

**Reply by Juuxel** [reply]
Yeah, that's probably the best solution for you.

Looking a bit closer, it seems that when you right-click/use a Croptopia "leaf crop block", it simulates a block breaking. Spectrum assumes that block breaking only happens on the logical server, but Croptopia also calls it on the logical client.

**Reply by rdyEmber** [reply]
wait, but i did not use "leaf crop block"
i was walking on surface killing animals

**Comment by warjort** [comment]
Looks like a known issue that has already been fixed?

https://github.com/DaFuqs/Spectrum/issues/134
https://github.com/DaFuqs/Spectrum/commit/7c1cd4dd6641ff078a33a7d2f5d63fdb3d11a06c

**Reply by rdyEmber** [reply]
did not interacted with any blocks, i was just walking on the surface killing animals

**Reply by warjort** [reply]
Your crash report says otherwise:
````at net.minecraft.class_636.method_2896(class_636.java:327)````

Which is the client side of player interactBlock()
https://github.com/FabricMC/yarn/blob/04cda4a2bb287e17f814b3e04a233df2b0e35b68/mappings/net/minecraft/client/network/ClientPlayerInteractionManager.mapping#L19

**Reply by rdyEmber** [reply]
strange, but i did not interact with any kind of block for real
i was holding Iron Twinblade and was slicing animals for meat

**Reply by warjort** [reply]
Maybe you accidently right clicked the ground?

**Reply by rdyEmber** [reply]
nope, i remember that when i killed a pig i looked up in the sky and thats where my minecraft crashed afterwards
